### ISSUE TITLE
In case of multiple bugs in same line, only first occurrence was detected

### ISSUE THREAD

**Issue by ysknmt** [spotbugs/spotbugs#1257]
When analyzing the following sample code, SpotBugs detect a BugInstance that has only **one** occurrence, even if I expected that SpotBugs detects a BugInstance that has **two** occurrences.

```java
class Sample {
  public String getString() {
    return new String("a") + new String("b");  // This line has two DM_STRING_CTOR
  }
}
```

Is this intended work? or limitations?

## My insight

The BugInstance for the second bug is also generated internally, but the BugAccumulator seems to throw it away due to the same hash value of the SourceLineAnnotation as that of the first bug.
https://github.com/spotbugs/spotbugs/blob/master/spotbugs/src/main/java/edu/umd/cs/findbugs/BugAccumulator.java#L108

**Comment by smallmass** [comment]
Hi, spotbugs team. 
When developing a spotbug plugin, I also encountered a similar situation to this case.
If it is inteded, could you present any reason if it is possible?

**Comment by JuditKnoll** [comment]
@smallmass AFAIK this is the intended behavior for BugAccumulator, for some bugtypes depending on the implementation, it can be really useful. When you implement a detector in spotbugs itself or in a plugin, you can use BugReporter, which doesn't accumulate the reported bugs, or using BugAccumulator you can report the BugInstance when finding it via `reportBug(BugInstance, Data)` or call `reportAccumulatedBugs()` before the second bugInstance is accumulated.

For the original issue with `DM_STRING_CTOR`, it needs to be checked, whether using BugAccumulator is really needed there.

**Comment by smallmass** [comment]
@JuditKnoll Thank you for your kindness. It is very helpful. :)
